packages/cozy-stack-client/src/NotesCollection.js

```diff
@@ -0,0 +1,54 @@
+import DocumentCollection from './DocumentCollection'
+import { uri } from './utils'
+
+export const NOTES_DOCTYPE = 'io.cozy.notes'
+
+const normalizeDoc = DocumentCollection.normalizeDoctypeJsonApi(NOTES_DOCTYPE)
+const normalizeNote = note => ({
+  ...normalizeDoc(note),
+  ...note.attributes
+})
+
+/**
+ * Implements `DocumentCollection` API to interact with the /notes endpoint of the stack
+ */
+class NotesCollection extends DocumentCollection {
+  constructor(stackClient) {
+    super(NOTES_DOCTYPE, stackClient)
+  }
+
+  /**
+   * all - Fetches all notes
+   *
+   * @returns {{data, links, meta}} The JSON API conformant response.
+   */
+  async all() {
+    //TODO: add support for pagination
+    const resp = await this.stackClient.fetchJSON('GET', '/notes')
+
+    return {
+      ...resp,
+      data: resp.data.map(normalizeNote)
+    }
+  }
+
+  /**
+   * destroy - Destroys the note on the server
+   *
+   * @param {io.cozy.notes} note     The note document to destroy
+   * @param {string}   note._id The note's id
+   *
+   * @returns {{ data }} The deleted note
+   */
+  async destroy({ _id }) {
```

Comment:
Is the note really deleted or is it just marked as deleted ? Why do we have to return the doc with _deleted ?

Reply:
It is really deleted, but without the `_deleted` flag the document won't be removed from queries; I Was under the impression this was the way to do it, [for example here](https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/DocumentCollection.js#L258).

Reply:
I had forgotten about that, thanks !